src/PostProcesses/motionBlurPostProcess.ts

```diff
@@ -181,6 +187,98 @@ export class MotionBlurPostProcess extends PostProcess {
         super.dispose(camera);
     }
 
+    /**
+     * Called on the mode changed (object based or screen based).
+     */
+    private _applyMode(): void {
+        if (!this._geometryBufferRenderer && !this._prePassRenderer) {
+            // We can't get a velocity or depth texture. So, work as a passthrough.
+            Logger.Warn("Multiple Render Target support needed to compute object based motion blur");
+            return this.updateEffect();
+        }
+
+        this._updateEffect();
+
+        this._invViewProjection = null;
+        this._previousViewProjection = null;
+
+        if (this.isObjectBased) {
+            if (this._prePassRenderer && this._prePassEffectConfiguration) {
+                this._prePassEffectConfiguration.texturesRequired[0] = Constants.PREPASS_VELOCITY_TEXTURE_TYPE;
+            }
+
+            this.onApply = (effect: Effect) => this._onApplyObjectBased(effect);
+        } else {
+            this._invViewProjection = Matrix.Identity();
+            this._previousViewProjection = Matrix.Identity();
+
+            if (this._prePassRenderer && this._prePassEffectConfiguration) {
+                this._prePassEffectConfiguration.texturesRequired[0] = Constants.PREPASS_DEPTHNORMAL_TEXTURE_TYPE;
+            }
+
+            this.onApply = (effect: Effect) => this._onApplyScreenBased(effect);
+        }
+    }
+
+    /**
+     * Called on the effect is applied when the motion blur post-process is in object based mode.

```

Comment:
missing doc for param effect

Reply:
Thanks! This is a private method but I can add if needed :)